src/libraries/Microsoft.Extensions.Hosting.Systemd/src/SystemdHostBuilderExtensions.cs

```diff
@@ -27,27 +27,63 @@ public static class SystemdHostBuilderExtensions
         public static IHostBuilder UseSystemd(this IHostBuilder hostBuilder)
         {
             if (SystemdHelpers.IsSystemdService())
             {
                 hostBuilder.ConfigureServices((hostContext, services) =>
                 {
-                    services.Configure<ConsoleLoggerOptions>(options =>
-                    {
-                        options.FormatterName = ConsoleFormatterNames.Systemd;
-                    });
-
-                    // IsSystemdService() will never return true for android/browser/iOS/tvOS
-#pragma warning disable CA1416 // Validate platform compatibility
-                    services.AddSingleton<ISystemdNotifier, SystemdNotifier>();
-                    services.AddSingleton<IHostLifetime, SystemdLifetime>();
-#pragma warning restore CA1416 // Validate platform compatibility
+                    AddSystemdLifetime(services);
                 });
             }
             return hostBuilder;
         }
+
+        /// <summary>
+        /// Configures the lifetime of the <see cref="IHost"/> built from <paramref name="services"/> to
+        /// <see cref="SystemdLifetime"/>, provides notification messages for application started
+        /// and stopping, and configures console logging to the systemd format.
+        /// </summary>
+        /// <remarks>
+        ///   <para>
+        ///     This is context aware and will only activate if it detects the process is running
+        ///     as a systemd Service.
+        ///   </para>
+        ///   <para>
+        ///     The systemd service file must be configured with <c>Type=notify</c> to enable
+        ///     notifications. See <see href="https://www.freedesktop.org/software/systemd/man/systemd.service.html"/>.
+        ///   </para>
+        /// </remarks>
+        /// <param name="services">
+        /// The <see cref="IServiceCollection"/> used to build the <see cref="IHost"/>.
+        /// For example, <see cref="HostApplicationBuilder.Services"/> or the <see cref="IServiceCollection"/> passed to the
+        /// <see cref="IHostBuilder.ConfigureServices(System.Action{HostBuilderContext, IServiceCollection})"/> callback.
+        /// </param>
+        /// <returns>The <paramref name="services"/> instance for chaining.</returns>
+        public static IServiceCollection AddSystemd(this IServiceCollection services)
+        {
+            if (SystemdHelpers.IsSystemdService())
+            {
+                AddSystemdLifetime(services);
+            }
+            return services;
+        }
+
+        private static void AddSystemdLifetime(IServiceCollection services)
+        {
+            services.Configure<ConsoleLoggerOptions>(options =>
+            {
+                options.FormatterName = ConsoleFormatterNames.Systemd;
+            });
+
+            // IsSystemdService() will never return true for android/browser/iOS/tvOS
+#pragma warning disable CA1416 // Validate platform compatibility
```

Comment:
The Windows service code below uses asserts for suppression. Should we do the same here instead of pragmas and comments?

Reply:
The issue here is that the unix condition is harder to write. The assert for windows is that it "is windows". The assert here would need to be not any of the following:

*    [UnsupportedOSPlatform("android")]
*    [UnsupportedOSPlatform("browser")]
*    [UnsupportedOSPlatform("ios")]
*    [UnsupportedOSPlatform("maccatalyst")]
*    [UnsupportedOSPlatform("tvos")]

Which isn't that bad. But where it does get bad is that you can't express this condition outside of NETCOREAPP since we don't have APIs in `RuntimeInformation` and the `OperatingSystem.IsXXX` methods are only available on NETCOREAPP.

So we need to pragma disable here.